### DISCUSSION TITLE
Batch/Group undo and redo changes

### DISCUSSION THREAD

**Discussion by pcjaarvis** [GrapesJS/grapesjs#6208] [Q&A]
Has anyone managed to batch or group changes?
Lets say I have 5 changes which happen in  sequence, I would like to group them somehow so that when I press undo all 5 are done at the same time, same for redo.

I couldn't figure out anything using the UndoManager, guess I'll have to extend it somehow.

Another approach I thought of was storing the whole editor state after each change, and manage my own stack of changes, but seems very expensive if the there are lots of components.

**Comment by elfalem** [comment]
I don't think there is a direct way to do this. However, changes are grouped by a property called `magicFusionIndex`. Having the same number for this property indicates a set of changes are in one group and can be undone/redone at once.

`editor.UndoManager.getStack().models[stackIndex].set('magicFusionIndex', x);`
